xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -102,6 +113,16 @@ protected void handleMessage(JsonNode message) {
                     "Channel {} has been failed: {}", channelName, statusMessage.getErrorMessage());
             }
             break;
+          case addOrderStatus:
```

Comment:
It may make more sense for these type of websocket transactions to keep as much of their logic within the service they reside in as possible. In this case keeping as much of the streaming trade logic inside `KrakenStreamingTradeService` so as to keep purpose of `KrakenStreamingService` clear and the footprint smaller.

I am thinking possibly use the `KrakenStreamingService` like a router where when raw messages that should be processed by say `KrakenStreamingTradeService` are send to that service by this switch statement so that `KrakenStreamingService`'s job really just becomes receiving messages to send and routing the result to the handling class.

@badgerwithagun usually has good ideas regarding implementation, not sure what you think?

Reply:
I agree that the Kraken trading logic should reside in KrakenStreamingTradeService and KrakenStreamingService should serve generic Kraken message handling - and this is how this is implemented. The logic that you see int the KrakenStreamingService is for queueing the messages/dispatching the statuses to proper requests and I believe this part should stay there, while generating the messages and actually handling them is in the KrakenStreamingTradeService

Reply:
sounds good I will defer to @badgerwithagun for this